xdsl/transforms/lower_mpi.py

```diff
@@ -742,7 +743,11 @@ def walker(op: Operation):
                                       len(module.body.blocks[0].ops))
 
 
-def lower_mpi(ctx: MLContext, module: builtin.ModuleOp):
+@dataclass
+class LowerMPIPass(OperationPass):
+
+    name = 'lower-mpi'
+
     # TODO: how to get the lib info in here?
     lib_info = MpiLibraryInfo()
```

Comment:
```suggestion
    lib_info: MpiLibraryInfo = field(default_factory=MpiLibraryInfo)
```

Reply:
This will fix my issue for the MpiLibraryInfo, but the walkers are still an issue, AFAIU.